### DISCUSSION TITLE
Scheduled workflows trigger deployments

### DISCUSSION THREAD

**Discussion by jirikopecky** [community/community#67728] [Actions]
### Select Topic Area

Bug

### Body

We have scheduled workflow (`on.schedule` triggered) which runs periodic tests against deployed environment. To access the environment we need environment-specific variables and also correct context to authenticate with Azure using workload identity. This means specifying environment for the workflow job.

This creates deployment event, but contrary to workflows triggered by other events with jobs tied to environment, this deployment event triggers workflow.

This causes problems with our deployment workflows that are triggered by deployment event forcing us to add new conditions to the workflow (to check if all required parameters are passed in payload) and also causes a lot of inactive deployments being present in deployment view.

**Comment by Sachin066** [comment]
It appears that you're encountering a bug in your workflow system where a scheduled workflow, which is supposed to run periodic tests against a deployed environment, triggers a deployment event instead of behaving as expected. This unexpected behavior is causing issues with your deployment workflows and results in inactive deployments being visible in the deployment view.
